app/src/ui/diff/side-by-side-diff.tsx

```diff
@@ -1042,7 +993,7 @@ function highlightParametersEqual(
     (newProps === prevProps ||
       (newProps.file.id === prevProps.file.id &&
         newProps.showSideBySideDiff === prevProps.showSideBySideDiff)) &&
-    newState.document.diff.text === prevState.document.diff.text
+    newState.diff.text === prevState.diff.text
```

Comment:
I _think_ this now needs to be updated to return false when the `fileContents` change. As it stands now highlighting will not be applied for the first mount since at mount time the `fileContents` will be `null` and when it later gets loaded the check in [componentDidUpdate](https://github.com/desktop/desktop/blob/448c788f762d9ed49ef838912a83cae878e2b3fa/app/src/ui/diff/side-by-side-diff.tsx#L235-L240) will prevent `initDiffSyntaxMode` from running again. Perhaps something like this?

```suggestion
    newState.diff.text === prevState.diff.text &&
    prevProps.fileContents?.file.id === newProps.fileContents?.file.id
```

Reply:
You're totally right, nice catch again! ❤️